teps/0114-custom-tasks-beta.md

```diff
@@ -208,6 +209,19 @@ Remove guarding of `Custom Tasks` behind `enable-custom-tasks` and `enable-api-f
 When [TEP-0096: Pipelines V1 API][tep-0096] is implemented to add V1 API, `Custom Tasks` will be gated behind feature
 gate `enable-api-fields` being set to `"beta"` - this is out of scope for this TEP  (in scope for TEP-0096).
 
+##### Cancellation
+
+Implementing cancellation is now `required` instead of `optional`. If the Custom Task implementor does not support cancellation via `.spec.status`, `Pipeline` timeouts and `PipelineRun` cancellation will not work properly.
```

Comment:
What does `properly` allude to here?

@ScrapCodes please share your experience here if you have any

Reply:
To ensure that cancellation is required, are we planning to have validations now or in future? if no, is this just a documentation change?

Also, do we support pipeline level Timeout for custom task? I remember it was removed? 

What are the downside of keeping as: "It is upto the custom task controller to support it." i.e. optional, e.g. not all custom task may want to support it e.g. CEL. Those custom task who do not wish to support they are responsible for not supporting timeout as well as of today.

Today `Tekton controller` does not depend on custom task's support of cancel, is it going to change? 

>`Pipeline` timeouts and `PipelineRun` cancellation will not work properly. 
Can you elaborate?

Reply:
@pritidesai 
> What does properly allude to here?

PiplineRun controller has hard code dependencies on status report of CustomRun controller: See [the code](https://github.com/tektoncd/pipeline/blob/38c739a5d0d9faff8c362462f8b6beee0458d978/pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go#L258-L260), which [is used to decide if the CustomRun is failed or not](https://github.com/tektoncd/pipeline/blob/38c739a5d0d9faff8c362462f8b6beee0458d978/pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go#L148).

@ScrapCodes 
> To ensure that cancellation is required, are we planning to have validations now or in future? if no, is this just a documentation change?

Yes I think we'll have code validations in the future, I opened an issue for it https://github.com/tektoncd/pipeline/issues/5700

> do we support pipeline level Timeout for custom task? I remember it was removed?

There are two kinds of timeout for Custom Task:
- One is [`PipelineTask.Timeout`](https://github.com/tektoncd/pipeline/blob/b15a80e5ab9299386024578ac25b1f42c4d364ac/pkg/apis/pipeline/v1beta1/pipeline_types.go#L229), which is passed to `CustomRunSpec.Timeout`.
- The other is [`PipelineRunSpec.Timeouts`](https://github.com/tektoncd/pipeline/blob/b15a80e5ab9299386024578ac25b1f42c4d364ac/pkg/apis/pipeline/v1beta1/pipelinerun_types.go#L247). 

What we removed is the first one. i.e. we stopped PipelineRun controller to modify CustomRun.status.Conditions upon CustomRun timeout -- (CustomRun executes more than `CustomRunSpec.Timeout`, see https://github.com/tektoncd/pipeline/pull/5658).